components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -397,3 +406,78 @@ function OrgDefaultWorkspaceImageModal(props: OrgDefaultWorkspaceImageModalProps
         </Modal>
     );
 }
+
+interface WorkspaceClassOptionsProps {
+    settings: OrganizationSettings | undefined;
+    disabled: boolean;
+}
+const WorkspaceClassOptions = (props: WorkspaceClassOptionsProps) => {
+    const [errorMsg, setErrorMsg] = useState("");
+    const [selectedValue, setSelectedValue] = useState(props.settings?.allowedWorkspaceClasses ?? []);
+    const updateTeamSettings = useUpdateOrgSettingsMutation();
+    const { data: classes, isError, isLoading } = useWorkspaceClasses();
+
+    const { toast } = useToast();
+    const handleUpdateTeamSettings = useCallback(
+        async (classes: string[]) => {
+            try {
+                await updateTeamSettings.mutateAsync(
+                    {
+                        ...props.settings,
+                        allowedWorkspaceClasses: classes,
+                    },
+                    {
+                        onSuccess: () => {
+                            toast({ message: "Available workspace classes updated for your organization." });
+                        },
+                    },
+                );
+            } catch (error) {
+                if (!ErrorCode.isUserError(error["code"])) {
+                    console.error(error);
+                }
+                setErrorMsg(error.message);
+            }
+        },
+        [updateTeamSettings, props, toast],
+    );
+
+    if (isError || !classes) {
+        return <div>Something went wrong</div>;
+    }
+
+    if (isLoading) {
+        return <LoadingState />;
+    }
+
+    return (
+        <>
+            {errorMsg.length > 0 && (
+                <Alert type="error" className="mb-2">
+                    {errorMsg}
+                </Alert>
+            )}
+            <Label>
+                {classes.map((wsClass) => (
+                    <CheckboxInputField
+                        id={wsClass.id}
+                        label={wsClass.displayName}
+                        hint={wsClass.description}
+                        checked={selectedValue.includes(wsClass.id)}
+                        onChange={(checked) => {
```

Comment:
It would be great to make this consistent with the Configuration detail pages and have a <kbd>Save</kbd> button here instead of autosaving. It may be a bit unexpected if the user is familiar with those UIs.

Reply:
Use a `save` button can also save traffic

cc @corygrunk